MmSupervisorPkg/Test/MmSupvRequestUnitTestApp/MmSupvRequestUnitTestApp.c

```diff
@@ -843,6 +843,55 @@ InspectSecurityPolicy (
   return UNIT_TEST_PASSED;
 }
 
+/*
+  Test case to request communication buffer update from supervisor
+*/
+UNIT_TEST_STATUS
+EFIAPI
+RequestUpdateCommBuffer (
+  IN UNIT_TEST_CONTEXT  Context
+  )
+{
+  EFI_STATUS                        Status;
+  MM_SUPERVISOR_REQUEST_HEADER      *CommBuffer;
+  MM_SUPERVISOR_COMM_UPDATE_BUFFER  *UpdateCommBuffer;
+
+  // Grab the CommBuffer and fill it in for this test
+  Status = MmSupvRequestGetCommBuffer (&CommBuffer);
+  UT_ASSERT_NOT_EFI_ERROR (Status);
+
+  CommBuffer->Signature = MM_SUPERVISOR_REQUEST_SIG;
+  CommBuffer->Revision  = MM_SUPERVISOR_REQUEST_REVISION;
+  CommBuffer->Request   = MM_SUPERVISOR_REQUEST_COMM_UPDATE;
+  CommBuffer->Result    = EFI_SUCCESS;
+
+  // It really should not matter, but just in case...
+  // We intentionally set it to some bogus address so that it would fail
+  // catastrophically if the flow does not work.
+  UpdateCommBuffer                                                                        = (MM_SUPERVISOR_COMM_UPDATE_BUFFER *)(CommBuffer + 1);
+  UpdateCommBuffer->NewMmCoreData.MemoryDescriptor.PhysicalStart                          = 0xDEADBEEF;
+  UpdateCommBuffer->NewCommBuffers[MM_USER_BUFFER_T].MemoryDescriptor.PhysicalStart       = (EFI_PHYSICAL_ADDRESS)(UINTN)RequestUpdateCommBuffer;
+  UpdateCommBuffer->NewCommBuffers[MM_SUPERVISOR_BUFFER_T].MemoryDescriptor.PhysicalStart = (EFI_PHYSICAL_ADDRESS)(UINTN)RequestUpdateCommBuffer;
+
+  // This should cause the system to reboot.

```

Comment:
Wouldn't a reset in the middle of a test be  a problem?

Reply:
Good catch, bad copy and paste..

Reply:
Updated.